internal/olm/operator/registry/fbcindex/fbc_registry_pod.go

```diff
@@ -315,25 +346,51 @@ func (f *FBCRegistryPod) podForBundleRegistry(cs *v1alpha1.CatalogSource) (*core
 					Ports: []corev1.ContainerPort{
 						{Name: defaultContainerPortName, ContainerPort: f.GRPCPort},
 					},
-					VolumeMounts: volumeMounts,
+					VolumeMounts: sharedVolumeMounts,
 				},
 			},
 			ServiceAccountName: f.cfg.ServiceAccount,
 		},
 	}
 
+	if f.cfg.GzipCM {
+		f.addGZIPInitContainer(sharedVolumeMounts, gzipVolumeMount)
+	}
+
 	return f.pod, nil
 }
 
-// container creation command for FBC type images.
-const fbcCmdTemplate = `opm serve {{ .FBCIndexRootDir}} -p {{ .GRPCPort }}`
+func (f *FBCRegistryPod) addGZIPInitContainer(containerVolumeMount []corev1.VolumeMount, gzipVolumeMount []corev1.VolumeMount) {
+	initContainerVolumeMount := append(containerVolumeMount, gzipVolumeMount...)
+	f.pod.Spec.InitContainers = []corev1.Container{
```

Comment:
Nit: Is the init containers slice already initialized before this? Only concern I have here being that if we added init containers somewhere else in the future this would overwrite them all to just be the gzip init container rather than appending it.